custom_components/vaillant_vsmart/climate.py

```diff
@@ -125,7 +125,7 @@ def hvac_action(self) -> HVACAction:
             return HVACAction.OFF
 
         try:
-            if self._module.measured.temperature < self._module.measured.setpoint_temp:
+            if self._module.boiler_status == True:
                 return HVACAction.HEATING
         except TypeError:
             pass
```

Comment:
This try-except block was here because both `temperature` and `setpoint_temp` were optional - if they were `None` the check would fail.

I suggest removing the block completely and just checking if `self._module.boiler_status` is truthy:

```suggestion
        if self._module.boiler_status:
            return HVACAction.HEATING
```

Reply:
ok